src/ivcam/sr300-fw-update-device.cpp

```diff
@@ -22,6 +23,20 @@ namespace librealsense
         std::this_thread::sleep_for(std::chrono::seconds(10));
     }
 
+    bool sr300_update_device::check_fw_compatibility(const std::vector<uint8_t>& image) const
+    {
+        std::string fw_version = extract_firmware_version_string((const void*)image.data(), image.size());
+        auto device_pid = _usb_device->get_info().pid;
+        auto min_max_fw_it = device_to_fw_min_max_version.find(device_pid);
+        if (min_max_fw_it == device_to_fw_min_max_version.end())
+            throw std::runtime_error(to_string() << "Min and Max firmware versions have not been defined for this device: " << device_pid);
+
+        // advanced SR3XX devices do not fit the "old" fw versions and 
+        // legacy SR3XX devices do not fit the "new" fw versions
+        return (firmware_version(fw_version) >= firmware_version(min_max_fw_it->second.first)) &&
```

Comment:
In case the condition doesn't hold - print to log the version number extracted from the FW image

Reply:
added